core/modules/wconfig/WorkerConfig.h

```diff
@@ -36,65 +36,125 @@ namespace qserv {
 
 namespace wconfig {
 
-/// The Config object provides a thin abstraction layer to shield code from
-/// the details of how the qserv worker is configured.
+/**
+ *  Provide all configuration parameters for a Qserv worker instance
+ *
+ *  Parse an INI configuration file, identify required parameters and ignore
+ *  others, analyze and store them inside private member variables, use default
+ *  values for missing parameters, provide accessor for each of these variable.
+ *  This class hide configuration complexity
+ *  from other part of the code. All private member variables are related to INI
+ *  parameters and are immutables.
+ *
+ */
 class WorkerConfig {
 public:
-    WorkerConfig(util::ConfigStore const& ConfigStore);
 
+    /**
+     *  Create WorkerConfig instance from a INI configuration file
+     *
+     *  @param configFileName: path to worker INI configuration file
+     */
+    WorkerConfig(std::string configFileName)
+        : WorkerConfig(util::ConfigStore(configFileName)) {
+    }
+
+    WorkerConfig(WorkerConfig const&) = delete;
+    WorkerConfig& operator=(WorkerConfig const&) = delete;
 
-    /* Get groupScheduler group size
-     * set from configuration file
+    /* Get thread pool size for shared scans
+     *
+     * @return thread pool size for shared scans
      */
-    const unsigned int getThreadPoolSize() const {
+    unsigned int const getThreadPoolSize() const {
```

Comment:
`const` in return type does not do anything, please drop it here and everywhere below